pkg/storage/snapshot/restore.go

```diff
@@ -968,11 +956,17 @@ func (ctrl *VMRestoreController) createRestorePVC(
 	if err != nil {
 		return err
 	}
+	if volumeSnapshot == nil {
```

Comment:
@mhenriks Do you think this is physically possible without malicious-like actions by the user?

Reply:
imho we should ensure that https://github.com/kubevirt/kubevirt/blob/356f5376b7e990ef105c4dbbf385c1f56e26a37d/pkg/storage/snapshot/snapshot_base.go#L668 never return (`nil`, `nil`)

Reply:
@dankenigsberg It is a very common pattern in this project for functions that are wrappers around informers/indexes to return `nil, nil` if the resource does not exist.

Reply:
@akalenyu it is possible that VolumeSNapshots are deleted by admins for malicious reasons probably to free up storage on the back end

Reply:
@mhenriks thanks. Is it a kubevirt thing, or is it commonplace in other k8s controllers?

I (and probably the author of the modified code) find it it very surprising for a getter function to get nothing yet declare success.